### ISSUE TITLE
destroyStrategy isn't followed when a field is removed from DOM.

### ISSUE THREAD

**Issue by jbsaff** [Textalk/angular-schema-form#508]
http://schemaform.io/examples/bootstrap-example.html#/976bfc96771600614451

Toggle the checkbox, and the condition-limited field remains in the model.

Without the new builder:
If the condition is applied to a container (such as the deprecated "conditional" type, or a field set), then the fields are cleaned up following the appropriate destroyStrategy. But fields that have a condition applied directly as a form option have the $destroy listener set on the bootstrap-decorator wrapper tag, which is never actually removed from the DOM by conditions.

With the new builder: 
I'm seeing some fields work as expected (triggering to show a text field when a certain value in a drop down list is selected), but checkbox list fields don't appear to be following the condition behavior. I'll need to dig into this more for 0.8.5+ development.

**Comment by davidlgj** [comment]
Ohhh... interesting! Thanks for testing the new builder. I'll check it out. In the new builder the `sf-field` directive should take care of that. I'll try to check it out soon.

**Comment by jbsaff** [comment]
Okay, thanks for the follow-up. It sounds like the sf-field directive is what we talked about in #371. Is that for the new builder only? My quick perusal through the code seemed to indicate that was a possibility.

**Comment by davidlgj** [comment]
@jbsaff No its not quite what we talked about in #371. But you are right it's for the new builder. The `sf-field` directive replaces some of the things that the `bootstrap-decorator` (or any other decorator directive) does. Most importantly create a scope and set the appropriate form definition on it under the name `form` so that the template can use it. It's is added to first child of the template for each field.

**Comment by jbsaff** [comment]
Found the primary problem with checkboxes in the new builder: 
https://github.com/Textalk/angular-schema-form-bootstrap/blob/develop/src/bootstrap-decorator.js#L43

Condition isn't in the dependency list, so it's never checked for the checkboxes type. It looks like it was removed in commit e24d164534e4d6d872e14b0e6f5d9a1b37541dfc. I've added line comments in a couple places in each project related to this issue.
